satpy/readers/seviri_base.py

```diff
@@ -193,11 +193,28 @@
 
 
 def get_cds_time(days, msecs):
-    """Get the datetime object of the time since epoch given in days and
-    milliseconds of day
+    """Compute timestamp given the days since epoch and milliseconds of the day
+
+    Args:
+        days (int or numpy.ndarray):
+            Days since 1958-01-01
+        msecs (int or numpy.ndarray):
+            Milliseconds of the day
+
+    Returns:
+        numpy.datetime64: Timestamp(s)
+
     """
-    return datetime(1958, 1, 1) + timedelta(days=float(days),
-                                            milliseconds=float(msecs))
+    if isinstance(days, (int, float)):
+        days = np.array([days], dtype='int64')
+        msecs = np.array([msecs], dtype='int64')
+
+    time = np.datetime64('1958-01-01').astype('datetime64[ms]') + \
+        days.astype('timedelta64[D]') + msecs.astype('timedelta64[ms]')
+
+    if len(time) == 1:
+        return time[0]
```

Comment:
Is this needed ? Don't we always have the time as an array with multiple values .?

Reply:
Yes. the scanline timestamps are always multiple values. The use case I had in mind are attributes which contain a single timestamp `(day, msec)` (not sure whether they exist). If scalars are provided to the function, it would return a scalar, not an array. But I agree that it might be confusing to have different return types.